staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -232,3 +232,14 @@ func HasObjectMetaSystemFieldValues(meta Object) bool {
 	return !meta.GetCreationTimestamp().Time.IsZero() ||
 		len(meta.GetUID()) != 0
 }
+
+// ResetObjectMetaForStatus forces the meta fields for a status update to match the meta fields
+// for a pre-existing object.  Status updates are for updating status, not annotations
+func ResetObjectMetaForStatus(meta, existingMeta Object) {
+	meta.SetGeneration(existingMeta.GetGeneration())
+	meta.SetSelfLink(existingMeta.GetSelfLink())
+	meta.SetLabels(existingMeta.GetLabels())
+	meta.SetAnnotations(existingMeta.GetAnnotations())
```

Comment:
> Otherwise controllers that want to do this need to send two updates and wouldn't be able to make an atomic change.

This doesn't stop it from ever being possible (you can do something different the strategy as opposed to https://github.com/kubernetes/kubernetes/pull/45552/files#diff-fefac67143bac6db5964a9cb79e6c988R97), but it would change it to be non-default.  I think its more the exception than the rule.